src/__experimental__/components/switch/switch.spec.js

```diff
@@ -171,6 +179,38 @@ describe('Switch', () => {
     });
   });
 
+  describe('check icon when validating', () => {
+    const wrapper = render({
+      label: 'My Label',
+      labelHelp: 'Please help me?',
+      unblockValidation: true,
+      useValidationIcon: true
+    }, mount);
+    const validationTypes = ['error', 'warning', 'info'];
+
+    beforeEach(() => {
+      const props = {
+        tooltipMessage: 'The message',
+        hasError: false,
+        hasWarning: false,
+        hasInfo: false
+      };
+
+      wrapper.setProps(props);
+    });
+
+    describe.each(validationTypes)('validation %s', (type) => {
+      it('check icon type', () => {
+        const propName = `has${text.titleCase(type)}`;
+        wrapper.setProps({
+          [propName]: true
+        });
+
+        expect(wrapper.find(StyledValidationIcon).prop('data-validation-type')).toEqual(type);
```

Comment:
As we're trying to test the icon here I think we should be making assertions on the style and not the prop. The prop that we're asserting here isn't used in the styled component. So we're not really testing anything, from what I can see.

Can we update the `it` description so it says something along the lines of `validation error > displays error icon` instead of `validation error > check icon type`

Reply:
Updated in commit ee2c98595